src/modules/menu/components/search/SearchResultList.vue

```diff
@@ -65,31 +69,32 @@ defineExpose({ focusFirstEntry })
         >
             <div class="search-results-inner">
                 <SearchResultCategory
-                    v-show="results?.locationResults?.length > 0"
+                    v-show="locationResults.length > 0"
                     ref="locationCategory"
                     :title="i18n.t('locations_results_header')"
-                    :results="results.locationResults"
+                    :results="locationResults"
                     data-cy="search-results-locations"
                     @first-entry-reached="emit('firstResultEntryReached')"
                     @last-entry-reached="gotToLayerCategory()"
                 />
                 <SearchResultCategory
-                    v-show="results?.layerResults?.length > 0"
+                    v-show="layerResults.length > 0"
                     ref="layerCategory"
                     :title="i18n.t('layers_results_header')"
-                    :results="results.layerResults"
+                    :results="layerResults"
                     data-cy="search-results-layers"
-                    @show-layer-legend-popup="showLayerLegend"
                     @first-entry-reached="gotToLocationCategory()"
                 />
+                <SearchResultCategory
+                    v-show="layerFeatureResults.length > 0"
+                    ref="layerFeatureResults"
+                    :title="i18n.t('featuresearch_results_header')"
+                    :results="layerFeatureResults"
+                    data-cy="search-results-layer-features"
+                    @first-entry-reached="gotToLayerFeaturesCategory()"
+                />
             </div>
```

Comment:
Here the idea of the `first-entry-reached` and `last-entry-reached` is for the keyboard navigation so that for the user it can go up and down with the arrow key as if it had only one list, with this modification this is not the case anymore.
You should have something like this
- `first-entry-reached` => go to last entry of previous category or the search input field when no previous category
- `last-entry-reached` => got to next category first entry or nothing if no further category